airflow/providers/dbt/cloud/hooks/dbt.py

```diff
@@ -0,0 +1,488 @@
+
+    @classmethod
+    def check_is_valid(cls, statuses: Union[int, Sequence[int], Set[int]]):
+        """Validates input statuses are a known value."""
+        if isinstance(statuses, (Sequence, Set)):
+            for status in statuses:
+                cls(status)
+        else:
+            cls(statuses)
+
+    @classmethod
+    def is_terminal(cls, status: int) -> bool:
+        """Checks if the input status is that of a terminal type."""
+        cls.check_is_valid(statuses=status)
+
+        return status in cls.TERMINAL_STATUSES.value
+
+
+class DbtCloudJobRunException(AirflowException):
+    """An exception that indicates a job run failed to complete."""
+
+
+class DbtCloudHook(HttpHook):
+    """
+    Interact with dbt Cloud using the V2 API.
+
+    :param dbt_cloud_conn_id: The ID of the :ref:`dbt Cloud connection <howto/connection:dbt-cloud>`.
+    """
+
+    conn_name_attr = "dbt_cloud_conn_id"
+    default_conn_name = "dbt_cloud_default"
+    conn_type = "dbt_cloud"
+    hook_name = "dbt Cloud"
+
+    @staticmethod
+    def get_ui_field_behaviour() -> Dict[str, Any]:
+        """Builds custom field behavior for the dbt Cloud connection form in the Airflow UI."""
+        return {
+            "hidden_fields": ["host", "port", "schema", "extra"],
+            "relabeling": {"login": "Account ID", "password": "API Token"},
+        }
+
+    def __init__(self, dbt_cloud_conn_id: str = default_conn_name, *args, **kwargs) -> None:
+        super().__init__(auth_type=TokenAuth)
+        self.dbt_cloud_conn_id = dbt_cloud_conn_id
+        self.base_url = "https://cloud.getdbt.com/api/v2/accounts/"
+
+    @cached_property
+    def conn(self) -> Connection:
+        _conn = self.get_connection(self.dbt_cloud_conn_id)
+        if not _conn.password:
+            raise AirflowException("An API token is required to connect to dbt Cloud.")
+
+        return _conn
+
+    def get_conn(self, *args, **kwargs) -> Session:
+        session = Session()
+        session.auth = self.auth_type(self.conn.password)
+
+        return session
+
+    def _paginate(self, endpoint: str, payload: Optional[Dict[str, Any]] = None) -> List[Response]:
+        results = []
+        response = self.run(endpoint=endpoint, data=payload)
+        resp_json = response.json()
+        limit = resp_json["extra"]["filters"]["limit"]
+        num_total_results = resp_json["extra"]["pagination"]["total_count"]
+        num_current_results = resp_json["extra"]["pagination"]["count"]
+        results.append(response)
+
+        if not num_current_results == num_total_results:
+            _paginate_payload = payload.copy() if payload else {}
+            _paginate_payload["offset"] = limit
+
+            while True:
+                if num_current_results < num_total_results:
+                    response = self.run(endpoint=endpoint, data=_paginate_payload)
+                    resp_json = response.json()
+                    if resp_json["data"]:
```

Comment:
would it help to have log in the "else" if data is not present in the response.

Reply:
Certainly couldn't hurt. Perhaps this is better as a debug statement though since it's more related to the aspects of the API response rather than the task execution.

Reply:
I removed this check. The `data` object has no bearing on pagination. Thanks for the catch!